src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemAligner.java

```diff
@@ -200,14 +200,15 @@ public BwaMemIndex getIndex() {
                 nSequences += 1;
                 bufferCapacity += func.apply(ele).length + 1; // sequence length bytes + 1 for the trailing null
             }
-            final ByteBuffer contigBuf = ByteBuffer.allocateDirect(bufferCapacity);
+            final ByteBuffer contigBuf = BwaMemIndex.createByteBuffer(bufferCapacity);
             contigBuf.order(ByteOrder.nativeOrder());
             contigBuf.putInt(nSequences);
             for ( final T ele : iterable ) {
                 contigBuf.put(func.apply(ele)).put((byte) 0);
             }
             contigBuf.flip();
             alignsBuf = index.doAlignment(contigBuf, tmpOpts, pairEndStats);
+            BwaMemIndex.destroyByteBuffer(contigBuf);
```

Comment:
Can you free the buffer in the `finally` block instead?

Reply:
I had a PR two years ago (and so now useless) where I carefully checked the return value from every system call.  As long as I've reopened this can of worms I'll reapply the error checking.  That'll cover testing the malloc.
I almost put the buffer deletion into the finally.  I'll think a little more deeply about where it really belongs.
Thanks for stepping up to the plate.